### PR TITLE
support running test on mac for flutter branch workflow

### PR DESCRIPTION
**Pull request by eldrigo**

### CODE REVIEW

**Review by TechPiranja — APPROVED** [review]

File: .github/workflows/flutter-branch.yml [hunk 1]
```diff
@@ -17,6 +17,10 @@ on:
         description: "the directory to run each job of the workflow in"
         type: string
         default: "."
+      use-mac-runners-for-test:
+        description: "shoudl use mac runners to run test"
```
Comment (acpoppe) [inline .github/workflows/flutter-branch.yml]: NIT: typo here

File: .github/workflows/flutter-branch.yml [hunk 2]
```diff
@@ -49,7 +53,8 @@ jobs:
           working-directory: ${{ inputs.working-directory }}
       - run: flutter analyze
   test:
-    runs-on: [self-hosted]
+    runs-on:
+      ["${{ inputs.use-mac-runners-for-test && 'macOS' || 'self-hosted'  }}"]
```
Comment (acpoppe) [inline .github/workflows/flutter-branch.yml:57]: Self-hosted needs to be there either way.

` ["self-hosted"${{ inputs.use-mac-runners-for-test && ', "macOS" || ' ' }}]`

File: .github/workflows/flutter-branch.yml [hunk 3]
```diff
@@ -63,7 +68,8 @@ jobs:
       - run: flutter test --dart-define STAGE=mock
 
   test-coverage:
-    runs-on: [self-hosted]
+    runs-on:
+      ["${{ inputs.use-mac-runners-for-test && 'macOS' || 'self-hosted'  }}"]
```
Comment (acpoppe) [inline .github/workflows/flutter-branch.yml:72]: See above

**Review by acpoppe — CHANGES_REQUESTED** [review]

File: .github/workflows/flutter-branch.yml (hunk 2 above)
Comment (acpoppe) [reply .github/workflows/flutter-branch.yml:57]: You could also do

`[${{ inputs.use-mac-runners-for-test && ' "self-hosted", "macOS" ' || ' "self-hosted" ' }}]`

**Comment by eldrigo** [comment]
redundent.

**Pull request closed without merging**